numba_cuda/numba/cuda/testing.py

```diff
@@ -203,3 +209,96 @@ class ForeignArray(object):
     def __init__(self, arr):
         self._arr = arr
         self.__cuda_array_interface__ = arr.__cuda_array_interface__
+
+
+class FileCheckTestCaseMixin:
+    """
+    Mixin for tests that use FileCheck.
+
+    Methods assertFileCheckAsm and assertFileCheckLLVM will inspect a numba
+    CUDADispatcher and assert that the compmilation artifacts match the
+    FileCheck checks given in the kernel's docstring.
+
+    Method assertFileCheckMatches can be used to assert that a given string
+    matches FileCheck checks, and is not specific to CUDADispatcher.
+    """
+
+    def assert_filecheck_asm(
+        self,
+        ir_producer: CUDADispatcher,
```

Comment:
The `CUDADispatcher` type is not the same as the `CUDADispatcher` dispatcher object returned by the `cuda.jit` decorator.